Source/WebInspectorUI/UserInterface/Views/NavigationSidebarPanel.js

```diff
@@ -485,9 +485,11 @@ WI.NavigationSidebarPanel = class NavigationSidebarPanel extends WI.SidebarPanel
 
     pruneStaleResourceTreeElements()
     {
-        if (this._checkForStaleResourcesTimeoutIdentifier) {
-            clearTimeout(this._checkForStaleResourcesTimeoutIdentifier);
-            this._checkForStaleResourcesTimeoutIdentifier = undefined;
+        if (!this._checkForStaleResourcesTimeoutIdentifier) {
+            // Coalesce multiple calls to _checkForStaleResources happening this frame.
+            this._checkForStaleResourcesTimeoutIdentifier = setTimeout(() => {
+                this._checkForStaleResourcesTimeoutIdentifier = undefined;
+            });
         }
```

Comment:
this is fine, but i kinda feel like we should modify `Throttler` to support this 🤔

it'd probably involve adding a `fireOnceForThisTask` or something like that (similar to `Debouncer.prototype.delayForTime`)

feel free to do this as a followup if you want

Reply:
id imagine it'd do basically what you have this doing right now

the only reason we cant use `Throttler` today is that it's purely time based, but if we assume that every task is at most 16ms (i.e. 60fps) we probably do this and it'd work fine
```js
this._pruneStaleResourceTreeElementsThrottler = new Throttler(() => {
    this.pruneStaleResourceTreeElements();
}, 50);
```

anyways this was just me musing about how to bring this functionality to a more generalized utility since i can imagine it being useful for other things :)

i.e. what you have is totally fine